cmd/minishift/cmd/profile/profile_set.go

```diff
@@ -40,22 +36,44 @@ var profileSetCmd = &cobra.Command{
 func runProfile(cmd *cobra.Command, args []string) {
 	validateArgs(args)
 	profileName := args[0]
+	lastActiveProfile := profileActions.GetActiveProfile()
 
 	if !cmdUtil.IsValidProfile(profileName) {
+		if cmdUtil.GetVMStatus(lastActiveProfile) != "Does Not Exist" {
```

Comment:
That's a tad too many nested if statements here. Also there seems to be quite a bit of duplication between the branches. Can we please refactor this and break into multiple functions + removing some duplication.

Reply:
Sure